inspirehep/modules/workflows/tasks/magpie.py

```diff
@@ -0,0 +1,123 @@
+    if not base_url:
+        return
+    return "{base_url}/predict".format(
+        base_url=base_url
+    )
+
+
+def prepare_magpie_payload(record, corpus):
+    """Prepare payload to send to Magpie API."""
+    payload = dict(text="", corpus="keywords")
+    titles = filter(None, get_value(record, "titles.title", []))
+    abstracts = filter(None, get_value(record, "abstracts.value", []))
+    payload["text"] = ". ".join(
+        titles.encode('utf-8') + abstracts.encode('utf-8')
+    )
+    return payload
+
+
+def filter_magpie_response(labels, limit):
+    """Filter response from Magpie API, keeping most relevant labels."""
+    filtered_labels = []
+    for word, score in labels:
+        if score >= limit:
+            filtered_labels.append((word, score))
+
+    # In the event that there are no labels with a high enough score,
+    # we take only the top one
+    if labels and len(filtered_labels) == 0:
+        filtered_labels.append(labels[0])
+    return filtered_labels
+
+
+def guess_keywords(obj, eng):
+    """Workflow task to ask Magpie API for a keywords assessment."""
+    magpie_url = get_magpie_url()
+    if not magpie_url:
+        # Skip task if no API URL set
+        return
+    payload = prepare_magpie_payload(obj.data, corpus="keywords")
+    results = json_api_request(magpie_url, payload)
+    if results:
+        labels = results.get('labels', [])
+        keywords = filter_magpie_response(labels, limit=0.01)
+        cutoff = 10
+        obj.extra_data["keywords_prediction"] = dict(
+            labels=labels,
+            keywords=keywords[:cutoff]
+        )
+        current_app.logger.info("Keyword prediction (top 10): {0}".format(
+            obj.extra_data["keywords_prediction"]["keywords"]
+        ))
+
+
+def guess_categories(obj, eng):
+    """Workflow task to ask Magpie API for a subject area assessment."""
+    magpie_url = get_magpie_url()
+    if not magpie_url:
+        # Skip task if no API URL set
+        return
+    payload = prepare_magpie_payload(obj.data, corpus="categories")
+    results = json_api_request(magpie_url, payload)
+    if results:
+        labels = results.get('labels', [])
+        categories = filter_magpie_response(labels, limit=0.22)
+        obj.extra_data["categories_prediction"] = dict(
+            labels=labels,
+            categories=categories
+        )
+        current_app.logger.info("Category prediction: {0}".format(
+            obj.extra_data["categories_prediction"]["categories"]
+        ))
+
+
+def guess_experiments(obj, eng):
+    """Workflow task to ask Magpie API for a subject area assessment."""
+    magpie_url = get_magpie_url()
+    if not magpie_url:
+        # Skip task if no API URL set
+        return
+    payload = prepare_magpie_payload(obj.data, corpus="experiments")
+    results = json_api_request(magpie_url, payload)
+    if results:
+        labels = results.get('labels', [])
+        experiments = filter_magpie_response(labels, limit=0.5)
```

Comment:
yeah, I wanted to ask about that. I think it's fair, the confidence values for experiments are usually very high, because they are often explicitly mentioned in the abstract. You might want to extract it as a configuration constant or something, but not sure if it's necessary.